cmake/kokkos_tribits.cmake

```diff
@@ -213,46 +211,38 @@ MACRO(KOKKOS_PACKAGE_POSTPROCESS)
 ENDMACRO()
 
 MACRO(KOKKOS_MAKE_LIBKOKKOS)
-  IF (KOKKOS_SEPARATE_LIBS)
-    MESSAGE(FATAL_ERROR "Internal error: should not make single libkokkos with -DKOKKOS_SEPARATE_LIBS=On")
-  ENDIF()
-  IF(${CMAKE_VERSION} VERSION_LESS "3.12" OR MSVC)
-    #we are not able to set properties directly on object libraries yet
-    #so we have had to delay kokkos_link_options until here
-    #and do a bunch of other annoying work 
-    ADD_LIBRARY(kokkos ${KOKKOS_SOURCE_DIR}/core/src/dummy.cpp 
-      $<TARGET_OBJECTS:kokkoscore>
-      $<TARGET_OBJECTS:kokkoscontainers>
-    )
-    TARGET_LINK_LIBRARIES(kokkos PUBLIC ${KOKKOS_LINK_OPTIONS})
-    #still need the header-only library
-    TARGET_LINK_LIBRARIES(kokkos PUBLIC kokkosalgorithms)
-    KOKKOS_LINK_TPLS(kokkos)
-    #these properties do not work transitively correctly so we
-    #need some verbose hackery to make it work
-    GET_TARGET_PROPERTY(CORE_DIRS kokkoscore       INTERFACE_INCLUDE_DIRECTORIES)
-    GET_TARGET_PROPERTY(CTRS_DIRS kokkoscontainers INTERFACE_INCLUDE_DIRECTORIES)
-    TARGET_INCLUDE_DIRECTORIES(kokkos PUBLIC ${CORE_DIRS})
-    TARGET_INCLUDE_DIRECTORIES(kokkos PUBLIC ${CTRS_DIRS})
-
-    GET_TARGET_PROPERTY(CORE_FLAGS kokkoscore       INTERFACE_COMPILE_OPTIONS)
-    GET_TARGET_PROPERTY(CTRS_FLAGS kokkoscontainers INTERFACE_COMPILE_OPTIONS)
-    TARGET_COMPILE_OPTIONS(kokkos PUBLIC ${CORE_FLAGS})
-    TARGET_COMPILE_OPTIONS(kokkos PUBLIC ${CTRS_FLAGS})
-
-    GET_TARGET_PROPERTY(CORE_FEATURES kokkoscore       INTERFACE_COMPILE_FEATURES)
-    GET_TARGET_PROPERTY(CTRS_FEATURES kokkoscontainers INTERFACE_COMPILE_FEATURES)
-    TARGET_COMPILE_FEATURES(kokkos PUBLIC ${CORE_FEATURES})
-    TARGET_COMPILE_FEATURES(kokkos PUBLIC ${CTRS_FEATURES})
-  ELSE()
-    ADD_LIBRARY(kokkos ${KOKKOS_SOURCE_DIR}/core/src/dummy.cpp)
-    TARGET_LINK_LIBRARIES(kokkos PUBLIC kokkoscore kokkoscontainers)
-    TARGET_LINK_LIBRARIES(kokkos PUBLIC kokkosalgorithms)
-  ENDIF()
+  ADD_LIBRARY(kokkos ${KOKKOS_SOURCE_DIR}/core/src/dummy.cpp)
+  TARGET_LINK_LIBRARIES(kokkos PUBLIC kokkoscore kokkoscontainers)
+  TARGET_LINK_LIBRARIES(kokkos PUBLIC kokkosalgorithms)
```

Comment:
Any reason to not do it in one single `TARGET_LINK_LIBRARIES()` call?

Reply:
No, it's pre-existing. I am happy to change that in case we agree this is the approach we want.

Reply:
I had originally though this would be the way to go. This doesn't work if you're making a single static `libkokkos.a`. The code above would only create correct libraries if `kokkoscore`, etc, are CMake object libraries.